presto-iceberg/src/main/java/com/facebook/presto/iceberg/TableStatisticsMaker.java

```diff
@@ -497,34 +512,40 @@ private Optional<StatisticsFile> getClosestStatisticsFileForSnapshot(IcebergTabl
     /**
      * Builds a map of field ID to ColumnStatistics for a particular {@link StatisticsFile}.
      */
-    private Map<Integer, ColumnStatistics.Builder> loadStatisticsFile(StatisticsFile file)
+    private Map<Integer, ColumnStatistics> loadStatisticsFile(StatisticsFile file, StatisticsFileCache statisticsFileCache)
```

Comment:
I think you have a good point. I will consider creating a separate cache key which includes the table and columns as well. I think the StatisticsFile still needs to be in the cache key because different snapshots can reference different files. Open to more ideas for the cache key as well.

Reply:
The latest commit now uses a new cache key which is based off of the file and column ID. I didn't think table name was necessary since each file is specific to a table anyways. The logic for `loadStatisticsFile` has increased in complexity though. I would look more closely at that method during the next review.